fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2448,35 +2458,57 @@ private Bundle createSearchBundle(List<Resource> resources, FHIRSearchContext se
-        if (totalCount.getValue() < offset + pageSize) {
-            matchResourceCount = totalCount.getValue() - offset;
-        }
+        if (resources.size() > 0) {
+            // Calculate how many resources are 'match' mode
+            int pageSize = searchContext.getPageSize();
+            int offset = (searchContext.getPageNumber() - 1) * pageSize;
+            int matchResourceCount = pageSize;
+            if (totalCount.getValue() < offset + pageSize) {
+                matchResourceCount = totalCount.getValue() - offset;
+            }
 
-        // Check if too many included resources
-        if (resources.size() > matchResourceCount + SearchConstants.MAX_PAGE_SIZE) {
-            throw buildRestException(TOO_MANY_INCLUDE_RESOURCES, IssueType.BUSINESS_RULE, IssueSeverity.ERROR);
-        }
+            // Check if too many included resources
+            if (resources.size() > matchResourceCount + SearchConstants.MAX_PAGE_SIZE) {
+                throw buildRestException(TOO_MANY_INCLUDE_RESOURCES, IssueType.BUSINESS_RULE, IssueSeverity.ERROR);
+            }
 
-        for (Resource resource : resources) {
-            if (resource.getId() == null) {
-                throw new IllegalStateException("Returned resources must have an id.");
+            // Check for versioned references in 'match' resources for chained search
+            List<Issue> issues = new ArrayList<>();
+            List<Resource> matchResources = resources.subList(0,  matchResourceCount);
+            for (QueryParameter queryParameter : searchContext.getSearchParameters()) {
+                if (queryParameter.isChained() && !queryParameter.isReverseChained()) {
+                    // This is a chained search. Loop through results and generate an issue for any returned resource
+                    // that contains a versioned reference in the element associated with the chain search parameter.
+                    issues.addAll(checkForVersionedReference(ModelSupport.getResourceType(type), queryParameter, matchResources));
+                }
             }
-            // Search mode is determined by the matchResourceCount, which will be decremented each time through the loop.
-            // If the count is greater than 0, the mode is MATCH. If less than or equal to 0, the mode is INCLUDE.
-            Bundle.Entry entry = Bundle.Entry.builder()
-                    .fullUrl(Uri.of(getRequestBaseUri(type) + "/" + resource.getClass().getSimpleName() + "/" + resource.getId()))
-                    .resource(resource)
-                    .search(Search.builder()
-                        .mode(matchResourceCount-- > 0 ? SearchEntryMode.MATCH : SearchEntryMode.INCLUDE)
-                        .score(Decimal.of("1"))
-                        .build())
-                    .build();
 
-            bundleBuilder.entry(entry);
+            for (Resource resource : resources) {
+                if (resource.getId() == null) {
```

Comment:
Did you hit a point where this is true? It's a bit strange.

Reply:
I think he just carried it forward from what was there before.  see the old lines 2465 and 2466

Reply:
Correct. It was already there.